File: src/org/webfabric/web/servlet/OriginalPathInfo.java
package org.webfabric.web.servlet;

import org.webfabric.io.Path;

import javax.servlet.http.HttpServletRequest;

public class OriginalPathInfo implements Path {
    private final String value;

    public OriginalPathInfo(String value) {
        this.value = value;
    }

    public static OriginalPathInfo create(HttpServletRequest request) {
        return new OriginalPathInfo((String) request.getAttribute("javax.servlet.include.servlet_info"));
    }

    public String value() {
        return value;
    }

    @Override
    public String toString() {
        return value;
    }
}

File: src/org/webfabric/web/servlet/OriginalPathInfo.scala
package org.webfabric.web.servlet;



import io.{HierarchicalPath, Path}
import javax.servlet.http.HttpServletRequest;

class OriginalPathInfo(v: String) extends HierarchicalPath(v)

object OriginalPathInfo {
  def apply(request: HttpServletRequest):OriginalPathInfo = {
    return new OriginalPathInfo(request.getAttribute("javax.servlet.include.servlet_info").asInstanceOf[String]);
  }
}
